### PR TITLE
Usage examples

### PR DESCRIPTION
**Pull request by Mbah-Javis**

### CODE REVIEW

File: tensorflow/python/ops/image_ops_impl.py
Comment (kyscg) [inline tensorflow/python/ops/image_ops_impl.py]:
```suggestion
    ```python
```

File: tensorflow/python/ops/image_ops_impl.py
Comment (kyscg) [inline tensorflow/python/ops/image_ops_impl.py]:
```suggestion
    >>> x = tf.random.normal(shape=(256, 256, 3))
```

File: tensorflow/python/ops/image_ops_impl.py
Comment (kyscg) [inline tensorflow/python/ops/image_ops_impl.py]:
```suggestion
    >>> tf.image.transpose(x) 
```

Comment (kyscg) [inline tensorflow/python/ops/image_ops_impl.py]:
```suggestion
    tf.image.transpose(x) 
    ```
```

File: tensorflow/python/ops/image_ops_impl.py
Comment (kyscg) [inline tensorflow/python/ops/image_ops_impl.py]:
```suggestion
   
```

Comment (kyscg) [reply tensorflow/python/ops/image_ops_impl.py]: No need to import tf everywhere

File: tensorflow/python/ops/image_ops_impl.py
Comment (kyscg) [inline tensorflow/python/ops/image_ops_impl.py]:
```suggestion
    >>> x = tf.random.normal(shape=(256, 256, 3))
```

File: tensorflow/python/ops/image_ops_impl.py
Comment (kyscg) [inline tensorflow/python/ops/image_ops_impl.py]:
```suggestion
    >>> tf.image.rgb_to_yiq(x)
```

File: tensorflow/python/ops/image_ops_impl.py
Comment (kyscg) [inline tensorflow/python/ops/image_ops_impl.py]:
```suggestion
    
```

File: tensorflow/python/ops/image_ops_impl.py
Comment (kyscg) [inline tensorflow/python/ops/image_ops_impl.py]:
```suggestion
    >>> x = tf.random.normal(shape=(256, 256, 3))
```

File: tensorflow/python/ops/image_ops_impl.py
Comment (kyscg) [inline tensorflow/python/ops/image_ops_impl.py]:
```suggestion
    >>> tf.image.yiq_to_rgb(x)
```

File: tensorflow/python/ops/image_ops_impl.py
Comment (kyscg) [inline tensorflow/python/ops/image_ops_impl.py]:
```suggestion
  
```

File: tensorflow/python/ops/image_ops_impl.py
Comment (kyscg) [inline tensorflow/python/ops/image_ops_impl.py]:
```suggestion
  >>> x = tf.random.normal(shape=(256, 256, 3))
```

File: tensorflow/python/ops/image_ops_impl.py
Comment (kyscg) [inline tensorflow/python/ops/image_ops_impl.py]:
```suggestion
  >>> tf.image.yuv_to_rgb(x)
```

**Review by kyscg — CHANGES_REQUESTED** [review]
No need to import tf everywhere + improved consistency.

**Comment by Mbah-Javis** [comment]
[@kyscg](https://github.com/kyscg) what else is needed for this PR to be merged?

**Comment by kyscg** [comment]
We need a review from @alextp first. Maybe after the hols? I don't know

**Comment by Mbah-Javis** [comment]
Okay and thanks for your reply

File: tensorflow/python/ops/image_ops_impl.py
Comment (alextp) [inline tensorflow/python/ops/image_ops_impl.py]: Don't use a random image; instead use a small image (like a 1x2) and show the result so this can be a doctest

File: tensorflow/python/ops/image_ops_impl.py
Comment (alextp) [inline tensorflow/python/ops/image_ops_impl.py]: Likewise, no random image but instead something where the numbers are easy to understand

**Review by alextp — CHANGES_REQUESTED** [review]

File: tensorflow/python/ops/image_ops_impl.py
Comment (Mbah-Javis) [reply tensorflow/python/ops/image_ops_impl.py]: Please can I use an image from a url?

Comment (alextp) [reply tensorflow/python/ops/image_ops_impl.py]: I prefer a small image and showing the result as it doesn't require internet access on the testing harness

File: tensorflow/python/ops/image_ops_impl.py
Comment (yashk2810) [inline tensorflow/python/ops/image_ops_impl.py]: Please remove ```python and ``` (backticks) here and everywhere.

You can check out the instructions at https://www.tensorflow.org/community/contribute/docs_ref

Comment (Mbah-Javis) [reply tensorflow/python/ops/image_ops_impl.py]: all removed

File: tensorflow/python/ops/image_ops_impl.py
Comment (Mbah-Javis) [reply tensorflow/python/ops/image_ops_impl.py]: I have made the changes please check to see if its okay.

File: tensorflow/python/ops/image_ops_impl.py
Comment (yashk2810) [inline tensorflow/python/ops/image_ops_impl.py]: Please remove ```.

Also add the outputs. Please follow the guide I linked before to test.

File: tensorflow/python/ops/image_ops_impl.py
Comment (alextp) [inline tensorflow/python/ops/image_ops_impl.py]: This was built using tf 1.x not tf2; in tf2 tensors have values

Comment (Mbah-Javis) [reply tensorflow/python/ops/image_ops_impl.py]: Yap that's true Let me change it

Comment (Mbah-Javis) [reply tensorflow/python/ops/image_ops_impl.py]: The output in tf2 has numpy arrays should include it in the all i should add (```) before  and after the the code?

File: tensorflow/python/ops/image_ops_impl.py
Comment (Mbah-Javis) [reply tensorflow/python/ops/image_ops_impl.py]: The outputs has numpy arrays which if i copy and put them it won't look good. I suggest that i should add (```) as the guide says for code that can not be tested for some reasons. Please let me know your opinion thank you.

Comment (Mbah-Javis) [reply tensorflow/python/ops/image_ops_impl.py]: Or I should just allow the code like that?

Comment (yashk2810) [reply tensorflow/python/ops/image_ops_impl.py]: >  I suggest that i should add (```) as the guide says
No, please keep the doctest. 

> copy and put them it won't look good
It will look good.

Also, please use a smaller image size and if possible use tf.constant so that the value won't change on every run.

Comment (Mbah-Javis) [reply tensorflow/python/ops/image_ops_impl.py]: okay

Comment (Mbah-Javis) [reply tensorflow/python/ops/image_ops_impl.py]: a smaller image of about what size?

Comment (yashk2810) [reply tensorflow/python/ops/image_ops_impl.py]: Maybe (3, 3) or (2,2)?

Comment (Mbah-Javis) [reply tensorflow/python/ops/image_ops_impl.py]: I have made the changes and included the outputs

File: tensorflow/python/ops/image_ops_impl.py
Comment (yashk2810) [inline tensorflow/python/ops/image_ops_impl.py]: Please use `tf.constant`.

Random numbers don't help demonstrate what transpose really does. Also, doctests will fail on every run since random values will change.

So, I would suggest passing in an array and then using transpose on it and then showing the output.

Similarly comment for everywhere else.

File: tensorflow/python/ops/image_ops_impl.py
Comment (yashk2810) [inline tensorflow/python/ops/image_ops_impl.py]: Can you use an array like this:

image = [[1, 2, 3],
                 [4, 5, 6],
                 [7, 8, 9]]

and then show the output for this?

Also, remember to use `...` for line continuation. See the guide and please test it out locally so that you will see the errors.

File: tensorflow/python/ops/image_ops_impl.py
Comment (Mbah-Javis) [reply tensorflow/python/ops/image_ops_impl.py]: I have used `tf.constant` now they are no longer random numbers from the output of the transpose and others you will see that the shapes are different implying that it has been transposed and the output does not changes

File: tensorflow/python/ops/image_ops_impl.py
Comment (yashk2810) [reply tensorflow/python/ops/image_ops_impl.py]: Also, see the docstrings for `tf.concat` as an example.

Comment (Mbah-Javis) [reply tensorflow/python/ops/image_ops_impl.py]: the array you gave raises two value errors
ValueError: Shape (3, 3) must have rank at least 3
ValueError: 'image' must be at least three-dimensional.

Comment (Mbah-Javis) [reply tensorflow/python/ops/image_ops_impl.py]: I have seen it let me change the arrays i used

Comment (Mbah-Javis) [reply tensorflow/python/ops/image_ops_impl.py]: I have made the changes please see if there is any other thing to be done before this PR can be merged.

**Comment by Mbah-Javis** [comment]
@yashk2810  please is there any other thing that I need to do before this PR can be merged?

File: tensorflow/python/ops/image_ops_impl.py
Comment (alextp) [inline tensorflow/python/ops/image_ops_impl.py]: Use fewer digits of precision (so replace 2.399996 with 2.3....); same below

Comment (Mbah-Javis) [reply tensorflow/python/ops/image_ops_impl.py]: Okay. But before i will do that I will like to ask if it is proper to edit outputs? because those are  output array values not an input values

File: tensorflow/python/ops/image_ops_impl.py
Comment (yashk2810) [inline tensorflow/python/ops/image_ops_impl.py]: Can you use integer input values here?

Also, use a 2x2 matrix (here and everywhere else).

File: tensorflow/python/ops/image_ops_impl.py
Comment (Mbah-Javis) [reply tensorflow/python/ops/image_ops_impl.py]: > Can you use integer input values here?
> 
> Also, use a 2x2 matrix (here and everywhere else).

This can only be done with `tf.image.transpose()` and `tf.image.adjust_hue()`
all the others will raise ValueErrors

File: tensorflow/python/ops/image_ops_impl.py
Comment (Mbah-Javis) [reply tensorflow/python/ops/image_ops_impl.py]: > Can you use integer input values here?
> 
> Also, use a 2x2 matrix (here and everywhere else).

This can only be done with `tf.image.transpose()` and `tf.image.adjust_hue()`
all the others will raise ValueErrors

File: tensorflow/python/ops/image_ops_impl.py
Comment (yashk2810) [inline tensorflow/python/ops/image_ops_impl.py]: Use integer values and a 2-D array if possible here and everywhere else.

Comment (Mbah-Javis) [reply tensorflow/python/ops/image_ops_impl.py]: I have made the changes. But integer values and a 2D array isn't possible because 
of the kernel structures of those I haven't changed. The kernel looks like this
`_rgb_to_yiq_kernel = [[0.299, 0.59590059, 0.2115],
                      [0.587, -0.27455667, -0.52273617],
                      [0.114, -0.32134392, 0.31119955]]` and the others which i haven't changed are similar. So using a 2D array will raise a Matrix-size incompatible error

**Review by mihaimaruseac — APPROVED** [review]

File: tensorflow/python/ops/image_ops_impl.py
Comment (mihaimaruseac) [inline tensorflow/python/ops/image_ops_impl.py]: You will have to remove the empty lines inside the output (like this one)

File: tensorflow/python/ops/image_ops_impl.py
Comment (mihaimaruseac) [inline tensorflow/python/ops/image_ops_impl.py]: This needs removed

File: tensorflow/python/ops/image_ops_impl.py
Comment (mihaimaruseac) [inline tensorflow/python/ops/image_ops_impl.py]: This needs removal

File: tensorflow/python/ops/image_ops_impl.py
Comment (mihaimaruseac) [inline tensorflow/python/ops/image_ops_impl.py]: No need for empty line here

File: tensorflow/python/ops/image_ops_impl.py
Comment (mihaimaruseac) [inline tensorflow/python/ops/image_ops_impl.py]: This might need to be indented.

File: tensorflow/python/ops/image_ops_impl.py
Comment (mihaimaruseac) [inline tensorflow/python/ops/image_ops_impl.py]: This line is overindented

File: tensorflow/python/ops/image_ops_impl.py
Comment (mihaimaruseac) [inline tensorflow/python/ops/image_ops_impl.py]: Rmove this empty line

File: tensorflow/python/ops/image_ops_impl.py
Comment (mihaimaruseac) [inline tensorflow/python/ops/image_ops_impl.py]: Remove empty line.

File: tensorflow/python/ops/image_ops_impl.py
Comment (mihaimaruseac) [inline tensorflow/python/ops/image_ops_impl.py]: Remove empty line.

File: tensorflow/python/ops/image_ops_impl.py
Comment (mihaimaruseac) [inline tensorflow/python/ops/image_ops_impl.py]: Too much indent. Should start on the same column as `returns:` above

File: tensorflow/python/ops/image_ops_impl.py
Comment (mihaimaruseac) [inline tensorflow/python/ops/image_ops_impl.py]: Remove empty line.

File: tensorflow/python/ops/image_ops_impl.py
Comment (mihaimaruseac) [inline tensorflow/python/ops/image_ops_impl.py]: Remove empty line.

File: tensorflow/python/ops/image_ops_impl.py
Comment (mihaimaruseac) [inline tensorflow/python/ops/image_ops_impl.py]: Remove empty line.

File: tensorflow/python/ops/image_ops_impl.py
Comment (mihaimaruseac) [inline tensorflow/python/ops/image_ops_impl.py]: Remove empty line.

File: tensorflow/python/ops/image_ops_impl.py
Comment (mihaimaruseac) [inline tensorflow/python/ops/image_ops_impl.py]: Remove empty line.

File: tensorflow/python/ops/image_ops_impl.py
Comment (mihaimaruseac) [inline tensorflow/python/ops/image_ops_impl.py]: Remove empty line.

File: tensorflow/python/ops/image_ops_impl.py
Comment (mihaimaruseac) [inline tensorflow/python/ops/image_ops_impl.py]: pylint might complain about the lack of space after `,` in `],[`. If it does, please fix.

**Review by mihaimaruseac — CHANGES_REQUESTED** [review]

**Comment by Mbah-Javis** [comment]
@mihaimaruseac are the changes okay?

**Review by mihaimaruseac — APPROVED** [review]

**Comment by mihaimaruseac** [comment]
Can you please resolve conflicts and rebase on master?

**Comment by Mbah-Javis** [comment]
> Can you please resolve conflicts and rebase on master?

Ya I have resolved the conflicts

**Review by mihaimaruseac — APPROVED** [review]

**Comment by mihaimaruseac** [comment]
This will get rolled back as it breaks internal tests

```
Failed example:
    tf.image.rgb_to_yuv(image)
Expected:
    <tf.Tensor: shape=(3, 2, 3), dtype=float32, numpy=
    array([[[ 1.815     ,  0.5831516 , -0.7149856 ],
          [ 4.815     ,  0.5831516 , -0.7149855 ]],
         [[ 7.815     ,  0.5831516 , -0.7149856 ],
          [10.815001  ,  0.5831518 , -0.7149852 ]],
         [[13.815001  ,  0.58315134, -0.7149857 ],
          [16.815     ,  0.58315134, -0.7149854 ]]], dtype=float32)> 
Got:
    <tf.Tensor: shape=(3, 2, 3), dtype=float32, numpy=
    array([[[ 1.815     ,  0.5831516 , -0.7149856 ],
            [ 4.815     ,  0.5831516 , -0.7149856 ]],

           [[ 7.815     ,  0.5831516 , -0.7149856 ],
            [10.815001  ,  0.58315134, -0.7149854 ]],

           [[13.815     ,  0.58315134, -0.7149856 ],
            [16.815     ,  0.58315134, -0.7149854 ]]], dtype=float32)>
```

**Comment by Mbah-Javis** [comment]
@mihaimaruseac That should be because of the spacing between the arrays right?

**Pull request merged**
